lib/gollum-lib/page.rb

```diff
@@ -10,11 +10,18 @@ class << self
       #
       # query     - The String path to match.
       # entry     - The BlobEntry to check against.
-      def path_match(query, entry, global_match = false)
+      # global_match - If true, find a File matching path's filename, but not it's directory (so anywhere in the repo)
+      # sub_spaces   - GitHub compatibility: substitutes - for spaces when comparing filenames.
+      def path_match(query, entry, global_match = false, sub_spaces = false)
         return false if "#{entry.name}".empty?
         return false unless valid_extension?(entry.name)
-        entry_name = valid_extension?(query) ? entry.name : strip_filename(entry.name)        
-        query == (global_match ? ::File.join('/', entry_name) : ::File.join('/', entry.dir, entry_name))
+        entry_name = valid_extension?(query) ? entry.name : strip_filename(entry.name)     
+        match_path = ::File.join([
```

Comment:
This refactor is better than line 17 in the original anyway, even if we don't want the `sub_spaces` options.

Reply:
👍